azure-iot-device/azure/iot/device/common/mqtt_transport.py

```diff
@@ -264,14 +264,20 @@ def on_message(client, userdata, mqtt_message):
         mqtt_client.on_publish = on_publish
         mqtt_client.on_message = on_message
 
+        # Set paho automatic-reconnect delay to 2 hours.  Ideally we would turn
+        # paho auto-reconnect off entirely, but this is the best we can do.  Without
+        # this, we run the risk of our auto-reconnect code and the paho auto-reconnect
+        # code conflicting with each other.
+        mqtt_client.reconnect_delay_set(120 * 60)
```

Comment:
>reconnect_delay_set [](start = 20, length = 19)

any reason for 2 hrs ?

Reply:
2 hours is completely arbitrary.

---
In reply to: [399582990](https://github.com/Azure/azure-iot-sdk-python/pull/516#discussion_r399582990) [](ancestors = 399582990)